auth/user_mgt_test.go

```diff
@@ -0,0 +1,704 @@
+			},
+			PasswordHash: "passwordHash",
+			PasswordSalt: "passwordSalt",
+		},
+	}
+}
+
+type basicCompare struct {
+	got  interface{}
+	want interface{}
+}
+
+func TestGetUsers(t *testing.T) {
+	setListUsers()
+	s, closer := echoServer("list_users.json", t)
+	defer closer()
+	iter := s.Client().Users(context.Background(), "")
+	i := 0
+	for {
+		user, err := iter.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			t.Error(err)
+
+		}
+		tests := []basicCompare{
+			{user, listUsers[i]},
+			{user.CustomClaims, listUsers[i].CustomClaims},
+			{user.UserInfo, listUsers[i].UserInfo},
+			{user.UserRecord, listUsers[i].UserRecord},
+			{user.ProviderUserInfo, listUsers[i].ProviderUserInfo},
+			{user.UserMetadata, listUsers[i].UserMetadata},
+			{user.PasswordHash, listUsers[i].PasswordHash},
+			{user.PasswordSalt, listUsers[i].PasswordSalt},
+		}
+		for k, pui := range user.ProviderUserInfo {
+			tests = append(tests, basicCompare{pui, listUsers[i].ProviderUserInfo[k]})
+		}
+		for j, test := range tests {
+			if !reflect.DeepEqual(test.got, test.want) {
+				t.Errorf("item %d %d \ngot:    %T %#v \nwanted: %T %#v", i, j,
+					test.got, test.got, test.want, test.want)
+			}
+		}
+		i++
+	}
+
+}
+
+type badParamsTest struct {
+	params         *UserParams
+	expectingError string
+}
+
+func TestBadCreateUser(t *testing.T) {
+	badUserParams := []badParamsTest{
+		{
+			&UserParams{Password: ptr.String("short")},
+			"error in params: Password must be at least 6 chars long",
+		}, {
+			&UserParams{PhoneNumber: ptr.String("1234")},
+			"error in params: PhoneNumber must begin with a +",
+		}, {
+			&UserParams{PhoneNumber: ptr.String("+_!@#$")},
+			"error in params: PhoneNumber must contain an alphanumeric character",
+		}, {
+			&UserParams{UID: ptr.String("")},
+			"error in params: UID must not be empty",
+		}, {
+			&UserParams{UID: ptr.String(strings.Repeat("a", 129))},
+			"error in params: UID must be at most 128 chars long",
+		}, {
+			&UserParams{DisplayName: ptr.String("")},
+			"error in params: DisplayName must not be empty",
+		}, {
+			&UserParams{PhotoURL: ptr.String("")},
+			"error in params: PhotoURL must not be empty",
+		}, {
+			&UserParams{Email: ptr.String("")},
+			"error in params: Email must not be empty",
+		}, {
+			&UserParams{Email: ptr.String("a")},
+			"error in params: Email must contain exactly one '@' sign",
+		}, {
+			&UserParams{Email: ptr.String("a@")},
+			"error in params: Email must have non empty account and domain",
+		}, {
+			&UserParams{Email: ptr.String("@a")},
+			"error in params: Email must have non empty account and domain",
+		}, {
+			&UserParams{Email: ptr.String("a@a@a")},
+			"error in params: Email must contain exactly one '@' sign",
+		},
+	}
+	for i, test := range badUserParams {
+
+		_, err := client.CreateUser(context.Background(), test.params)
+		if err == nil {
+			t.Errorf("%d) expecting error %s", i, test.expectingError)
```

Comment:
Follow the error reporting conventions followed in existing tests. This should be `[%d] want error: %s; got nil`